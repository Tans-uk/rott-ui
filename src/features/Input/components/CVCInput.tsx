import {type FC} from 'react'

import {StyleSheet, TextInput} from 'react-native'

import {formatMessage, translator} from '../../../libs'
import {ThemeConfig} from '../../../models'
import {AlertDialog} from '../../AlertDialog'
import {Icon} from '../../Icon'
import {Item} from '../../Item'
import {Label} from '../../Label'
import {Pressable} from '../../Pressable'
import type {CVCInputProps} from '../models'
import {useCVCInputStyles, useInputStyles} from '../styles'

export const CVCInput: FC<CVCInputProps<ThemeConfig>> = ({
  fontSize,
  onChangeText,
  theme,
  size,
  ...props
}) => {
  const {defaultTextInputStyle} = useInputStyles({fontSize, theme, size})
  const {infoIcon} = useCVCInputStyles()
  const handleTextChange = (inputText: string) => {
    let text = inputText.replace(/[^0-9]/g, '')
    text = text.length > 3 ? text.substring(0, 3) : text
    onChangeText!(text)
  }

  return (
    <Item row>
      <TextInput
        editable={!props.disabled}
        placeholder='***'
        style={StyleSheet.flatten([defaultTextInputStyle])}
        keyboardType='number-pad'
        maxLength={3}
        onChangeText={handleTextChange}
        {...props}
      />

      {/* TODO: CVC Info butonuna kart arkayüzü eklenmeli */}
      <Pressable
        testID='info-icon-test-id'
        style={infoIcon}
        row
        justifyContentCenter
        alignItemsCenter
        onPress={() => {
          AlertDialog.show({
            title: formatMessage('CVC.INFO.TITLE'),
            text: formatMessage('CVC.INFO.DESCRIPTION'),
            buttons: {
              cancelButton: {
                text: 'COMMON.OK',
                variant: 'primary',
                onPress: () => AlertDialog.hide(),
              },
            },
          })
        }}>
        <Label variant='grey-200' fontSize='xs' fontWeight={'700'}>
          {translator('CVC.LABEL.DESCRIPTION')}
        </Label>
        <Icon name='CHECK_CIRCLE' height={18} width={18} variant='grey-200' marginLeft={4} />
      </Pressable>
    </Item>
  )
}
