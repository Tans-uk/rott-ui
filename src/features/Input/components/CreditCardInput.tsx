import React, {type FC} from 'react'

import {StyleSheet} from 'react-native'

import type {CreditCardInputProps} from '../models'
import {InputStyles} from '../styles'
import {InputContainer} from './InputContainer'

// Package Imports
import MaskInput from 'react-native-mask-input'

export const CreditCardInput: FC<CreditCardInputProps> = ({
  fontSize,
  onChangeText,
  theme,
  disabled,
  size,
  ...props
}) => {
  const MASK = [
    /\d/,
    /\d/,
    /\d/,
    /\d/,
    '-',
    /\d/,
    /\d/,
    /\d/,
    /\d/,
    '-',
    /\d/,
    /\d/,
    /\d/,
    /\d/,
    '-',
    /\d/,
    /\d/,
    /\d/,
    /\d/,
  ]

  const handleOnChangeText = (text: string) => {
    const replacedText = text.replace(/[^0-9]/g, '')

    onChangeText!(replacedText)
  }

  return (
    <InputContainer {...props} size={size} theme={theme}>
      <MaskInput
        editable={!disabled}
        mask={MASK}
        placeholder='**** **** **** ****'
        keyboardType='number-pad'
        maxLength={19}
        onChangeText={(_masked, unmasked) => handleOnChangeText(unmasked)}
        style={StyleSheet.flatten([InputStyles({fontSize, theme, size}).defaultTextInputStyle])}
        {...props}
      />
    </InputContainer>
  )
}
