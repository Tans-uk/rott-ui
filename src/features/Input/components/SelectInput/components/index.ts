export * from './SelectInput'
export * from './SelectInputModalComponent'
