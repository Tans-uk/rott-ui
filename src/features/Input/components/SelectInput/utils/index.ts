export * from './listHeightNormalizer'
export * from './modalHeightPercentageNormalizer'
export * from './sortListBySearchPriority'
