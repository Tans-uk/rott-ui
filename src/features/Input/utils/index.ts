export * from './formatByDateMode'
export * from './useInputStyleNormalizer'
