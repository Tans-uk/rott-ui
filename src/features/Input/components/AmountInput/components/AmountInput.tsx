import {useCallback, useContext, useEffect, useRef, useState, type FC} from 'react'

import {StyleSheet, TextInput} from 'react-native'

import {RottUiContext} from '../../../../../contexts'
import {ThemeConfig} from '../../../../../models'
import {Icon} from '../../../../Icon'
import {Item} from '../../../../Item'
import {Label} from '../../../../Label'
import {useInputStyles} from '../../../styles'
import {useInputStyleNormalizer} from '../../../utils'
import type {AmountInputProps} from '../models'
import {AmountInputStyles} from '../styles'

export const AmountInput: FC<AmountInputProps<ThemeConfig>> = ({
  fontSize,
  onChangeText,
  value,
  theme,
  disabled,
  size,
  currencyType = 'TL',
  ...props
}) => {
  const {defaultTextInputStyle} = useInputStyles({fontSize, theme, size})
  const {colors} = useContext(RottUiContext)
  const amountRef = useRef<TextInput>(null)
  const currencyRef = useRef<TextInput>(null)
  const [amount, setAmount] = useState('')
  const [currency, setCurrency] = useState('')

  const resetInternalStates = useCallback(() => {
    setAmount('')
    setCurrency('')
    onChangeText!('0.00')
  }, [])

  const replaceTextWithNumberOrEmpty = (text: string) => {
    const numericOnly = text.replace(/[^0-9]/g, '')

    if (numericOnly === '0') return numericOnly

    return numericOnly.replace(/^0+/, '')
  }

  const amountNormalizer = (text: string) => {
    if (!text) return resetInternalStates()

    const replacedAmount = replaceTextWithNumberOrEmpty(text)
    const reversedText = replacedAmount.split('').reverse().join('') ?? ''
    const dottedText = reversedText?.match(/.{1,3}/g)?.join('.')
    const normalizedText = dottedText?.split('').reverse().join('')

    setAmount(normalizedText ?? '')
  }

  const formatCurrency = (text: string) => {
    return text.replace(/[^0-9]/g, '')
  }

  const handleTextChange = (inputTextAmount: string, inputTextCurrency: string) => {
    if (!inputTextAmount && !inputTextCurrency) return

    let amountFormat = replaceTextWithNumberOrEmpty(inputTextAmount)
    let currencyFormat = formatCurrency(inputTextCurrency)

    amountFormat = amountFormat.length === 0 ? '0' : amountFormat

    currencyFormat = currencyFormat.length === 0 ? '00' : currencyFormat
    currencyFormat = currencyFormat.length === 1 ? `${currencyFormat}0` : currencyFormat

    if (currencyFormat.length > 2) currencyFormat = currencyFormat.substring(0, 2)

    onChangeText!(`${amountFormat}.${currencyFormat}`)
  }

  const placeholderColorNormalizer =
    theme === 'dark'
      ? colors.white
      : (!amount && !currency) ||
          ((amount === '0' || amount === '') && (!currency || currency === '' || currency === '00'))
        ? colors['grey-200']
        : colors['grey-900']

  useEffect(() => {
    const initialAmount = value?.split(',')[0]
    const initialCurrency = value?.split(',')[1]

    setAmount(initialAmount ?? '')
    setCurrency(initialCurrency ?? '')
  }, [])

  useEffect(() => {
    if (value === '' || !value || value === '0.00' || value === '0') resetInternalStates()
  }, [value])

  useEffect(() => {
    handleTextChange(amount, currency)
  }, [amount, currency])

  return (
    <Item relative onTouchStart={() => amountRef.current?.focus()}>
      <Item row {...props}>
        <TextInput
          ref={amountRef}
          nativeID='amount-native-id'
          testID='amount-test-id'
          editable={!disabled}
          placeholder='0'
          maxLength={11}
          value={amount}
          keyboardType='number-pad'
          onChangeText={(text) => amountNormalizer(text)}
          style={StyleSheet.flatten([defaultTextInputStyle, AmountInputStyles().amountInputStyle])}
          placeholderTextColor={placeholderColorNormalizer}
        />

        <Item
          justifyContentFlexEnd
          paddingBottom={useInputStyleNormalizer({size}).bottomElementPadding}>
          <Label
            fontSize={fontSize ?? useInputStyleNormalizer({size}).placeholderSize.toString()}
            fontWeight='bold'
            color={placeholderColorNormalizer as string}>
            ,
          </Label>
        </Item>

        <TextInput
          ref={currencyRef}
          testID='currency-test-id'
          maxLength={2}
          placeholder='00'
          value={currency}
          keyboardType='number-pad'
          onChangeText={(text) => {
            if (text.isEmpty()) amountRef.current?.focus()
            setCurrency(formatCurrency(text).substring(0, 2))
          }}
          onKeyPress={({nativeEvent}) => {
            if (nativeEvent.key === 'Backspace' && currency.isEmpty()) amountRef.current?.focus()
          }}
          onFocus={() => (currency === '00' ? setCurrency('') : undefined)}
          onBlur={() => (currency.isEmpty() || currency === '0' ? setCurrency('00') : undefined)}
          style={StyleSheet.flatten([defaultTextInputStyle, AmountInputStyles().amountInputStyle])}
          placeholderTextColor={placeholderColorNormalizer}
        />
      </Item>

      <Item absolute right={0} bottom={useInputStyleNormalizer({size}).icon.paddingBottom}>
        <Icon
          testID='currency-icon-test-id'
          name={currencyType ?? ''}
          width={useInputStyleNormalizer({size}).icon.width}
          height={useInputStyleNormalizer({size}).icon.height}
          color={colors['grey-200']}
        />
      </Item>
    </Item>
  )
}
