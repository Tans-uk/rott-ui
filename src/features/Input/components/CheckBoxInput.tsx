/* eslint-disable react-native/no-inline-styles */
import {type FC} from 'react'

import {StyleSheet} from 'react-native'

import {themeConfig} from '../../../providers'
import {Item} from '../../Item'
import {Label} from '../../Label'
import {Pressable} from '../../Pressable'
import {type CheckBoxInputProps} from '../models'
import {InputStyles} from '../styles'
import {InputStyleNormalizer} from '../utils'
import React from 'react'

export const CheckBoxInput: FC<CheckBoxInputProps> = ({
  description,
  checked,
  fontSize,
  theme,
  size,
  disabled,
  variant,
  fontWeight,
  onCheckChange,
  testID,
  ...props
}) => {
  return (
    <Pressable
      flex={0}
      testID={testID ?? 'checkbox-input-test-id'}
      onPress={() => !disabled && onCheckChange && onCheckChange(!checked)}
      {...props}>
      <Item
        row
        alignItemsCenter
        style={[
          StyleSheet.flatten([InputStyles({theme, size}).defaultTextInputStyle, {height: 'auto'}]),
        ]}
        testID='checkbox-container-test-id'>
        <Item
          borderRadius={8}
          borderColor={checked ? themeConfig.colors.primary : themeConfig.colors['grey-200']}
          borderWidth={2}
          width={24}
          height={24}
          alignItemsCenter
          justifyContentCenter>
          {(checked || disabled) && (
            <Item
              testID='checkbox-checked-test-id'
              borderWidth={2}
              borderColor={
                checked ? themeConfig.colors['neutral-blue-soft'] : themeConfig.colors['grey-200']
              }
              borderRadius={4}
              backgroundColor={
                checked ? themeConfig.colors.primary : themeConfig.colors['grey-200']
              }
              width={checked ? 15 : 22}
              height={checked ? 15 : 22}
            />
          )}
        </Item>

        {description && typeof description === 'string' && (
          <Item flex={1} style={{flexDirection: 'row'}}>
            <Label
              testID='checkbox-default-label-test-id'
              fontSize={fontSize ?? InputStyleNormalizer({size}).placeholderSize}
              fontWeight={fontWeight ?? 'normal'}
              marginLeft={12}
              variant={variant ?? (theme === 'dark' ? 'white' : 'grey-900')}
              style={{flexWrap: 'wrap'}}>
              {description}
            </Label>
          </Item>
        )}

        {description && typeof description !== 'string' && <>{description}</>}
      </Item>
    </Pressable>
  )
}
