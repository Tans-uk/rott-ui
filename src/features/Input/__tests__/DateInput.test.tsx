import React from 'react'

import {fireEvent, render, waitFor} from '../../../__tests__/utils/testUtils'
import {DateInput} from '../components'

import {startOfDay} from 'date-fns'

describe('Date Input -> Custom Input', () => {
  const dateInputTestId = 'date-input-test-id'
  const dateInputValueContainerTestId = 'date-input-value-container'
  const dateInputConfirmButtonTestId = 'date-input-confirm-button'
  const dateInputClearmButtonTestId = 'date-input-clear-button'
  const dateInputModalTestId = 'date-input-modal'
  const currentDate = new Date()

  it('ilk render anında snapshot ile eşleşmeli', () => {
    const renderedDateInput = render(
      <DateInput name='test' testID={dateInputTestId} date={currentDate} />
    )

    expect(renderedDateInput).toMatchSnapshot()
  })

  it('date inputa tıklandığında input modal olarak açılmalı', () => {
    const {getByTestId} = render(<DateInput name='test' date={currentDate} />)

    const dateInputValueContainer = getByTestId(dateInputValueContainerTestId)
    fireEvent.press(dateInputValueContainer)

    const dateInputModal = getByTestId(dateInputModalTestId)
    expect(dateInputModal).toBeVisible()
  })

  it('date input allowClear propertysi almadıysa değer temizleme butonu ekranda gözükmemeli.', () => {
    const {getByTestId, queryByTestId} = render(<DateInput name='test' date={currentDate} />)

    const dateInputValueContainer = getByTestId(dateInputValueContainerTestId)
    fireEvent.press(dateInputValueContainer)

    const clearButton = queryByTestId(dateInputClearmButtonTestId)
    expect(clearButton).toBeNull()
  })

  it('date input allowClear propertysi aldıysa Temizle butonuna tıklandığında değer temizlenmeli.', async () => {
    const onDateChangeMock = jest.fn()
    const {getByTestId, queryByTestId} = render(
      <DateInput
        name='test'
        value={currentDate.toDateString()} // Fix: Pass a valid date value in the format 'YYYY-MM-DD'.
        date={currentDate}
        mode='date'
        allowClear
        onDateChange={onDateChangeMock}
      />
    )

    const dateInputValueContainer = getByTestId(dateInputValueContainerTestId)
    fireEvent.press(dateInputValueContainer)
    const modalElement = getByTestId(dateInputModalTestId)
    expect(modalElement).toBeVisible()
    await waitFor(() => {
      const dateInput = getByTestId(dateInputTestId)
      expect(dateInput).toBeOnTheScreen()
      expect(dateInput).toHaveProp('date')
    })
    const clearButton = getByTestId(dateInputClearmButtonTestId)
    fireEvent.press(clearButton)

    const modalShouldNotVisible = queryByTestId(dateInputModalTestId)
    expect(modalShouldNotVisible).not.toBeOnTheScreen()
  })

  it('date input mode date olarak renderlanmalı', async () => {
    const {getByTestId} = render(<DateInput name='test' mode='date' />)

    const dateInputValueContainer = getByTestId(dateInputValueContainerTestId)
    fireEvent.press(dateInputValueContainer)

    await waitFor(() => {
      const dateInput = getByTestId(dateInputTestId)
      expect(dateInput).toBeOnTheScreen()
      expect(dateInput).toHaveProp('mode', 'date')
    })
  })

  it('date input mode time olarak renderlanmalı', async () => {
    const {getByTestId} = render(<DateInput name='test' mode='time' />)

    const dateInputValueContainer = getByTestId(dateInputValueContainerTestId)
    fireEvent.press(dateInputValueContainer)

    await waitFor(() => {
      const dateInput = getByTestId(dateInputTestId)
      expect(dateInput).toBeOnTheScreen()
      expect(dateInput).toHaveProp('mode', 'time')
    })
  })

  it('date input mode datetime olarak renderlanmalı', async () => {
    const {getByTestId} = render(<DateInput name='test' mode='datetime' />)

    const dateInputValueContainer = getByTestId(dateInputValueContainerTestId)
    fireEvent.press(dateInputValueContainer)

    await waitFor(() => {
      const dateInput = getByTestId(dateInputTestId)
      expect(dateInput).toBeOnTheScreen()
      expect(dateInput).toHaveProp('mode', 'datetime')
    })
  })

  it('minimum date verildikten sonra daha geçmiş bir gün seçilirse tanımlanan minimum date değer olarak atanmalı.', async () => {
    const onDateChangeMock = jest.fn()
    const {getByTestId} = render(
      <DateInput
        name='test'
        mode='date'
        date={new Date()}
        minimumDate={currentDate}
        onDateChange={onDateChangeMock}
      />
    )

    const dateInputValueContainer = getByTestId(dateInputValueContainerTestId)
    fireEvent.press(dateInputValueContainer)

    const dateInput = getByTestId(dateInputTestId)
    expect(dateInput).toBeOnTheScreen()
    await waitFor(() => {
      fireEvent(dateInput, 'onDateChange', new Date('2023.09.14'))

      setTimeout(() => {}, 150)
    })

    const confirmButton = getByTestId(dateInputConfirmButtonTestId)
    fireEvent.press(confirmButton)

    expect(onDateChangeMock).toHaveBeenCalledWith(startOfDay(currentDate))
  })

  it('maximum date verildikten sonra daha ileri bir gün seçilirse tanımlanan maximum date değer olarak atanmalı.', async () => {
    const onDateChangeMock = jest.fn(() => currentDate)
    const maxDate = new Date(currentDate)
    const {getByTestId} = render(
      <DateInput
        name='test'
        mode='date'
        date={currentDate}
        maximumDate={maxDate}
        onDateChange={onDateChangeMock}
      />
    )

    await waitFor(() => {
      const dateInputValueContainer = getByTestId(dateInputValueContainerTestId)

      fireEvent.press(dateInputValueContainer)
    })

    await waitFor(() => {
      const dateInput = getByTestId(dateInputTestId)
      expect(dateInput).toBeOnTheScreen()

      // Create a new date that's one day after the maximum date without mutating currentDate
      const futureDate = new Date(currentDate.getTime() + 24 * 60 * 60 * 1000)
      fireEvent(dateInput, 'onDateChange', futureDate)

      const confirmButton = getByTestId(dateInputConfirmButtonTestId)
      fireEvent.press(confirmButton)
    })

    expect(onDateChangeMock).toHaveBeenCalledWith(startOfDay(maxDate)) 
  })
})
