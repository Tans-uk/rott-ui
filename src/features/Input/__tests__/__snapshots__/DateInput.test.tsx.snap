// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date Input -> Custom Input ilk render anında snapshot ile eşleşmeli 1`] = `
<KeyboardProvider
  enabled={true}
>
  <View
    flex={1}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <View
      style={
        {
          "flex": undefined,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        animated={false}
        collapsable={false}
        flex={0}
        focusable={true}
        height={89}
        justifyContentCenter={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        size="full"
        style={
          {
            "color": "rgba(255, 255, 255, 1)",
            "flex": 0,
            "flexDirection": "column",
            "fontFamily": "Markpro-Medium",
            "fontSize": 18,
            "height": 89,
            "justifyContent": "center",
            "letterSpacing": 0.5,
            "maxHeight": 171,
            "width": "100%",
          }
        }
        testID="date-input-test-id"
      >
        <Text
          style={
            {
              "color": "rgba(161, 173, 175, 1)",
              "fontFamily": "Markpro-Medium",
              "fontSize": 16,
              "fontWeight": undefined,
              "letterSpacing": 0.5,
              "textAlign": "auto",
            }
          }
        >
          Lütfen bir tarih seçiniz
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        animated={false}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "flex": 1,
            "flexDirection": "column",
          }
        }
        testID="pressable-test-id"
      >
        <View
          absolute={true}
          bottom={16}
          right={0}
          style={
            {
              "bottom": 31,
              "flex": undefined,
              "flexDirection": "column",
              "position": "absolute",
              "right": 0,
            }
          }
        >
          <View
            style={
              {
                "flex": undefined,
                "flexDirection": "column",
              }
            }
          >
            <Text
              height={46}
              stroke="rgba(161, 173, 175, 1)"
              strokeWidth={2}
              testID="mock-icon"
              width={46}
            >
              MockIcon
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        [
          {
            "alignSelf": "center",
            "elevation": 999999,
            "flex": 0,
            "maxWidth": "100%",
            "position": "absolute",
            "zIndex": 999999,
          },
          {
            "flexDirection": "column-reverse",
            "justifyContent": "flex-start",
            "top": 30,
            "width": 750,
          },
        ]
      }
    >
      <View
        style={
          {
            "bottom": 0,
          }
        }
      >
        <RCTSafeAreaView />
      </View>
    </View>
    <View
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        [
          {
            "alignSelf": "center",
            "elevation": 999999,
            "flex": 0,
            "maxWidth": "100%",
            "position": "absolute",
            "zIndex": 999999,
          },
          {
            "bottom": 40,
            "flexDirection": "column",
            "justifyContent": "flex-end",
            "width": 750,
          },
        ]
      }
    >
      <View
        style={
          {
            "bottom": 0,
          }
        }
      >
        <RCTSafeAreaView />
      </View>
    </View>
  </View>
</KeyboardProvider>
`;
